### PR TITLE
Docker compatibility adjustments

### PR DESCRIPTION
**Pull request by tomjn**
This attempts to make our provisioners a little more comfortable during docker builds. Note that this PR is not adding docker support.

## Checks

* [ ] I've updated the changelog.
* [ ] I've tested this PR
* [x] This PR is for the `develop` branch not the `stable` branch.
* [ ] This PR is complete and ready for review.

### CODE REVIEW

**Comment by tomjn** [comment]
I've some other adjustments but they're experimental, these changes should benefit VirtualBox provisioning by fixing problems that don't prevent VVV from running under vagrant

File: provision/core/mariadb/provision.sh [hunk 1]
```diff
@@ -98,19 +95,20 @@ function mysql_setup() {
   chmod 0644 "/home/vagrant/.my.cnf"
   vvv_info " * Copied /srv/config/mysql-config/root-my.cnf          to /home/vagrant/.my.cnf"
 
-  check_mysql_root_password
+  if [ "${VVV_DOCKER}" != 1 ]; then
```
Comment (msaggiorato) [inline provision/core/mariadb/provision.sh:98]: This whole function would only run if we're not on docker, so i'd say this check is unnecessary.

File: provision/core/vvv/provision.sh [hunk 2]
```diff
@@ -71,7 +71,10 @@ function cleanup_vvv(){
   mv /tmp/hosts /etc/hosts
 }
 export -f cleanup_vvv
+
+if [ "${VVV_DOCKER}" != 1 ]; then
 vvv_add_hook finalize cleanup_vvv 15
```
Comment (msaggiorato) [inline provision/core/vvv/provision.sh:76]: Indentation?
```suggestion
  vvv_add_hook finalize cleanup_vvv 15
```

File: provision/core/wp-cli/provision.sh [hunk 3]
```diff
@@ -17,20 +17,26 @@ function wp_cli_setup() {
...
+    curl -s https://raw.githubusercontent.com/wp-cli/wp-cli/master/utils/wp-completion.bash -o /srv/config/wp-cli/wp-completion.bash
+    chown vagrant /srv/config/wp-cli/wp-completion.bash
   else
     vvv_info " * Updating wp-cli..."
     wp --allow-root cli update --nightly --yes
     vvv_success " * WP CLI Nightly updated"
   fi
-  vvv_info " * Installing WP CLI doctor sub-command"
-  wp --allow-root package install git@github.com:wp-cli/doctor-command.git
+  if [ "${VVV_DOCKER}" != 1 ]; then
```
Comment (msaggiorato) [inline provision/core/wp-cli/provision.sh:35]: Why execute this only on non docker? Is this incompatible or useless in docker?

File: provision/provision-helpers.sh [hunk 4]
```diff
@@ -85,10 +85,14 @@ export -f check_network_connection_to_host
...
 # @see check_network_connection_to_host
 function network_check() {
+  if [ "${VVV_DOCKER}" == 1 ]; then
+    return 0
+  fi
+
   # Make an HTTP request to ppa.launchpad.net to determine if
   # outside access is available to us. Also check the mariadb
   declare -a hosts_to_test=(
-    "https://ppa.launchpad.net"
```
Comment (msaggiorato) [inline provision/provision-helpers.sh:91]: Is there any reason for this change on this particular PR?

File: provision/tests/run-provisioners.sh [hunk 5]
```diff
@@ -1,11 +1,15 @@
 #!/bin/bash
+DEBIAN_FRONTEND=noninteractive
+TZ=Europe/London
```
Comment (msaggiorato) [inline provision/tests/run-provisioners.sh:3]: Is UTC available as a timezone? Or is this an exact equivalent?

**Review by msaggiorato — CHANGES_REQUESTED** [review]

File: provision/core/mariadb/provision.sh (hunk 1 above)
Comment (tomjn) [reply provision/core/mariadb/provision.sh:98]: I don't understand, the check prevents it running on docker, this is desirable behaviour

File: provision/provision-helpers.sh (hunk 4 above)
Comment (tomjn) [reply provision/provision-helpers.sh:91]: there is no  https ppa.launchpad.net, never was.  They only  provide TLS for private PPA's

File: provision/tests/run-provisioners.sh (hunk 5 above)
Comment (tomjn) [reply provision/tests/run-provisioners.sh:3]: This is copied from elsewhere, and just for the build of the docker image, I suspect `UTC` may work but I don't know

File: provision/core/wp-cli/provision.sh (hunk 3 above)
Comment (tomjn) [reply provision/core/wp-cli/provision.sh:35]: Docker image builds vs Docker container scripts. We should not consider `VVV_DOCKER` to be final, think of it as an experimental internal temporary thing that's necessary until a proper system is in place

**Pull request merged**